pkg/sql/type_change.go

```diff
@@ -791,6 +795,49 @@ func (t *typeSchemaChanger) canRemoveEnumValue(
 		query.WriteString(fmt.Sprintf("SELECT %s FROM [%d as t] WHERE", columns, ID))
 		firstClause := true
 		validationQueryConstructed := false
+
+		for _, idx := range desc.AllIndexes() {
+			if pred := idx.GetPredicate(); pred != "" {
+				foundUsage, err := findUsagesOfEnumValue(pred, member, typeDesc.ID)
+				if err != nil {
+					return err
+				}
+				if foundUsage {
+					return pgerror.Newf(pgcode.DependentObjectsStillExist,
+						"could not remove enum value %q as it is being used in a predicate of index %s",
+						member.LogicalRepresentation, &tree.TableIndexName{
+							Table: tree.MakeUnqualifiedTableName(tree.Name(desc.GetName())),
+							Index: tree.UnrestrictedName(idx.GetName()),
+						})
+				}
+			}
+			keyColumns := make([]catalog.Column, 0, idx.NumKeyColumns())
+			for i := 0; i < idx.NumKeyColumns(); i++ {
+				col, err := desc.FindColumnWithID(idx.GetKeyColumnID(i))
+				if err != nil {
+					return errors.AssertionFailedf(
+						"failed to find column with id %s (%d) in %s (%d)",
+						idx.GetKeyColumnName(i), idx.GetKeyColumnID(i), desc.GetName(), desc.GetID(),
```

Comment:
nit: A `return errors.WithAssertionFailure(err)` should be enough here. This shouldn't ever happen seeing that the descriptor has been validated at this point.

Reply:
Done.